pkgs/development/libraries/pcl/default.nix

```diff
@@ -47,7 +47,7 @@ stdenv.mkDerivation rec {
     libXt
   ]
   ++ lib.optionals stdenv.isDarwin [ Cocoa AGL ]
-  ++ lib.optionals withCuda [ cudatoolkit ];
+  ++ lib.optionals withCuda [ cudaPackages.cuda_nvcc ];
```

Comment:
`cuda_nvcc` should go in `nativeBuildInputs` (H'm, I'm even surprised this has worked for you with it in `buildInputs`)